pkg/apis/kops/containerdconfig.go

```diff
@@ -45,6 +47,19 @@ type ContainerdConfig struct {
 	Runc *Runc `json:"runc,omitempty"`
 	// SelinuxEnabled enables SELinux support
 	SeLinuxEnabled bool `json:"selinuxEnabled,omitempty"`
+	// Nri configures the Node Resource Interface.
+	Nri *NRIConfig `json:"nri,omitempty"`
+}
+
+type NRIConfig struct {
+	// Enable NRI support in containerd
+	Enabled *bool `json:"enabled,omitempty"`
+	// SocketPath is the path to the NRI socket to create for NRI plugins to connect to
+	SocketPath *string `json:"socketPath,omitempty"`
```

Comment:
Shouldn't there at least be a default? Why does this even need to be configurable?

Reply:
> Shouldn't there at least be a default? Why does this even need to be configurable?

Yeah actually we had another round of discussion with Krisztian about options we want to give to user and decided for now not expose the socket path option. If there are some users that show interest we can add it.


_Answering here for all the defaults related comments:_
The thing is there are different levels where those defaults get set. It is better to depend on the defaults provided by containerd and NRI, rather than introducing an additional layer of defaults here. For example, the defaults within containerd or NRI may evolve over time, and not relying on local defaults here mitigates the risk of inconsistencies in configuration values.

How I imagine the values being passed is: for example, if a user has not explicitly set the `nri.enabled` field in kOps, it is mirrored in the `disable` field within the containerd `config.toml` (i.e., unset). Subsequently, for NRI, an unset value in the 'disable` field defaults to its current value, which is to disable the NRI.